SConstruct

```diff
@@ -362,6 +362,30 @@ if len(mongoclientVersionComponents) not in (3,4):
     print("Error: client version most be of the form w.x.y[-rcz][-string]")
     Exit(1)
 
+
+# The Scons 'default' tool enables a lot of tools that we don't actually need to enable.
+# On platforms like Solaris, it actually does the wrong thing by enabling the sunstudio
+# toolchain first. As such it is simpler and more efficient to manually load the precise
+# set of tools we need for each platform.
+# If we aren't on a platform where we know the minimal set of tools, we fallback to loading
+# the 'default' tool.
+tools = ["unittest", "integration_test", "textfile"]
+
+def decide_platform_tools():
+    if windows:
+        # we only support MS toolchain on windows
+        return ['msvc', 'mslink', 'mslib', 'mssdk']
+    elif linux:
+        return ['gcc', 'g++', 'gnulink', 'ar']
+    elif solaris:
+        return ['gcc', 'g++', 'gnulink', 'ar']
+    elif darwin:
+        return ['gcc', 'g++', 'applelink', 'ar']
+    else:
+        return ["default"]
+
+tools += decide_platform_tools()
```

Comment:
Small subtlety:

I think SCons runs these tools in order. So, before, the 'default' tools would get loaded, then ours. This reverses that order. If we ever had a tool that wanted to manipulate flags *after* the toolchain tools were loaded, it might not work well. 

Can we reverse the order so that we load the 'minimized defaults' first, and then our custom tools?